src/types/timestamp.rs

```diff
@@ -56,23 +56,14 @@ pub enum Mantissa {
 /// Returns the first `num_digits` digits of the specified `value`.
 // This is used in Timestamp's implementation of [PartialEq].
 fn first_n_digits_of(num_digits: u32, value: u32) -> u32 {
-    let total_digits = number_of_digits(value);
+    let total_digits = super::num_decimal_digits_in_u64(value as u64) as u32;
     if total_digits <= num_digits {
         return value;
     }
     // Truncate the trailing digits
     value / 10u32.pow(total_digits - num_digits)
 }
 
-/// Returns the number of base-10 digits needed to represent `value`.
-fn number_of_digits(value: u32) -> u32 {
-    match value {
-        0 => 1,
-        1 => 1,
-        i => (i as f64).log10().ceil() as u32,
-    }
-}
```

Comment:
🗺️ This method was buggy and its usages have been replaced by calls to `num_decimal_digits_in_u64` above.